### ISSUE TITLE
v2 feedback

### ISSUE THREAD

**Issue by will-stone** [will-stone/plastic#22]
I'd like to have an issue thread for general feedback on the now released v2 beta. I'll tag all the contributors, pull-requesters, and issue loggers. Please download the latest version of Plastic and switch to v2 beta from the theme selection preferences. Check the changelog for details.

@pkazmier 
@kayteh 
@xgz123 
@ivancuric
@krokofant 
@jstheoriginal
@leiserfg

Thanks 😘

**Comment by krokofant** [comment]
@will-stone Where's the changelog? 🙈

**Comment by ivancuric** [comment]
![code](https://user-images.githubusercontent.com/2827047/40350647-b5797b0e-5daa-11e8-989b-9e6683b78920.png)

Missing the syntax highlighting in escaped template literals.

**Comment by will-stone** [comment]
@krokofant go to Plastic in VScode > Changelog tab. Or see `CHANGLEOG.md` in this repo.

@ivancuric You mean not turning `${` green? That was to keep the scope specificity low. I wasn't sure about that at first either but I've actually found it a nicer way to visually digest now. After all, the brackets are punctuation and the colour is consistent with all the other punctuation. I think it works well when you have larger block of text:
![image](https://user-images.githubusercontent.com/654103/40351526-862a663e-5da4-11e8-9396-637ce403f897.png)

**Comment by ivancuric** [comment]
You're right, looks definetly better in larger blocks.

**Comment by jstheoriginal** [comment]
I’ll be honest. I find this colour scheme harder on my eyes (I have issues with lower contrasts), specifically the blue. In v1 it was more of a brighter electric blue, while in v2 it’s darker and more muted against the background so it’s harder for me to read. 

https://s7.postimg.cc/4p38ea06z/7_D086_DA4-8869-4051-_A308-8152_DCE622_F3.jpg

**Comment by will-stone** [comment]
Agreed. I changed the blue yesterday and am not enjoying it today. Your comparison is incorrect though: the new blue is called Havelock Blue (check `config/colours.mjs` for all the colours and their names/hex-codes).

Original Plastic Cornflower Blue: `#61AFEF`
Solid Cornflower Blue: `#6495ED`
Current v2 Havelock Blue: `#5590D9`

It would have been nice to find a named blue closer to the original, but I've tried quite a few and none of them feel quite right. I'm happy to revert this to original Plastic's blue. How are you finding the other colours?

**Comment by jstheoriginal** [comment]
Ah ok makes sense as the blue did seem even darker than the comparison was showing, so that explains it. I’m not sure if making sure the colours used have an actual name be a criteria makes the most sense, since the most important thing is that it displays well.

I’ll let you know on the rest once I can use it more. I noticed he green wasn’t as bright, but it’s not a huge change so should be fine. 

Thanks Will!

**Comment by will-stone** [comment]
Oh, I totally agree; colours for the sake of naming is just silly, but I thought it provided a good base. I really like the text (which is also the punctuation colour) and variable colours now so they will definitely be staying. Maybe I'll put the colourful colours (if that makes sense) back to v1 colours but keep the rest. The main upgrade of v2 is the build and config system, anyway.

**Comment by jstheoriginal** [comment]
https://www.colorhexa.com/61afef

That site calls it “soft blue”.

**Comment by xgz123** [comment]
v2 is good. i hope the punctuation one also get updated too.

**Comment by will-stone** [comment]
@xgz123 read the changelog for info on the punctuation theme.

@jstheoriginal Thanks. Soft Blue seems to cover quite a big range on that site but I do like finding new tools like this. I'll probably stick with Chirag for the naming though, as Whiskey is called "Slightly desaturated orange" on that site 😆

**Comment by will-stone** [comment]
v.1.11.1 released which uses v1 syntax colours (shouldn't have messed with perfection!). I'm going to have a think about how this can be set-up to make it even easier for Plastic Forks to be made (then you'll get the de-prioritised punctuation back 👍 ).

**Comment by dok42** [comment]
Really love Plastic v1. 
I use the deprioritised punctuation and is almost perfect.
Some nights I feel like a darker background would be better for my eyes tho.

I'm having trouble finding a light dimmed theme with such perfection.

**Comment by will-stone** [comment]
Hi @dok42 The workbench (the background, text, border colours etc.) will be the same across all versions of Plastic. This is the bit that's fiddly and fairly time-consuming to keep on top of (as Microsoft release new theme-able areas). Therefore, my thinking for variations of Plastic (Plastic Forks) will be that themers will only need to change a set of syntax colours (that I've pre-categorised) and not have to worry about the rest (see https://github.com/will-stone/plastic/blob/master/theme.yaml for a sneak peek but bear in mind it's subject to change). To get to your point, the background colour will remain the same. Has your laptop got a _night mode_? For example, Mac's have got Night Shift, Window have Night Light. This will raise the temperature of the screen and save your eyes. I find this a nicer solution as I switch to the white-background of my browser.

**Comment by will-stone** [comment]
Oh, and that theme file doesn't build yet, so please don't anyone get too excited and start making Plastic Forks just yet :-)

**Comment by will-stone** [comment]
New version of v2 beta released. This includes the deprioritsed-punctuation variation.

I've gotten quite far with the compiler, please have a play but don't publish any themes just yet until I've ironed out the kinks. Here's the process so far; this is a first draft at instructions for creating your own Plastic Fork:

- Clone the repo (to your vscode extensions folder so you can test it)
- (there will be an init script here, that I haven't written yet, that will clear out any meta info that's specific to my version of Plastic: publisher ID, display name etc.)
- run `yarn` to install dependencies
- open `theme.yaml` and follow the instructions provided
- run `yarn build` to compile normal and deprioritised-punctuation themes into `out/`
- (in the future, the build script will update `package.json` automatically, but in the meantime you'll need to update the `contributes.themes.label`s manually with your theme name)

Do not edit anything in `src/`.

I'm thinking of having a gallery for Plastic Forks on the homepage. To qualify, the theme needs to have only edited the `theme.yaml` file. I'm going to create a couple once I've finished with the build and init scripts.

Please let me know if you have comments on this proposed process; I may have missed something glaringly obvious or something may not seem right. Let me know in a comment on this thread.

Thanks in advance to anyone who takes this early version for a whirl.

Last but not least, let me know of any comments about the v2 themes themselves. All being well, I'll probably publish them tomorrow evening; now that the colours used are v1 colours again, I will be overwriting the legacy themes, so this really is your last chance to speak up if anything is bothering you about the new ones.

**Comment by will-stone** [comment]
v2 released!

I've used the softer blue colour for `secondary`, the bright blue was a little too harsh.

I've just found this extension: https://marketplace.visualstudio.com/items?itemName=rafamel.subtle-brackets I highly recommend it. I've set these options:

```
"subtleBrackets.styles": {
    "global": {
      "borderColor": "#61AFEF",
      "borderWidth": "2px",
      "borderStyle": "none none solid none"
    }
},
```

Which changes the bracket matcher border to a little underline:

<img width="602" alt="image" src="https://user-images.githubusercontent.com/654103/40747056-1a72051c-6454-11e8-9b11-8457f8dbd311.png">
